### ISSUE TITLE
Invalid toc parsing with inline code blocks

### ISSUE THREAD

**Issue by sergeych** [vmg/redcarpet#664]
~~~
# H1

## H2a

1. List item 1

    ~~~sh
    # Inner formatted text
    ~~~

## H2b
~~~
"# Inner formatted text" is mistakenly treated as a heading. Moreover, resulting too is not valid:
~~~html
<ul id="toc">
    <li>
        <a href="#h1">H1</a>
        <ul>
            <li>
                <a href="#h2a">H2a</a>
                <ul>
                    <li>
                        <a href="#h2b">H2b</a>
                    </li>
                </ul>
            </li>
        </ul>
~~~
- first `<ul>` is not closed.

**Comment by nathan-appere** [comment]
I can trigger this bug with `"## Valid h2\nRandom text\n```\n# Invalid h1\n```"` but not with `"## Valid h2\n```\n# Invalid h1\n```"`.

**Comment by robin850** [comment]
Hello,

This is actually a duplicate of #451 which is fixed since Redcarpet 3.5.0.

I'm giving this a close ; thanks for reporting.
